### PR TITLE
Add bundling to umd2 module using webpack, more typescript, switch to yarn, convert mocha tests to commonjs

### PR DESCRIPTION
**Pull request by UgurGumushan**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "validate-steuerid",
-  "version": "1.0.2",
+  "version": "1.0.3",
```
Comment (DanaNussair) [inline package.json:3]: @UgurGumushan on npm the version is still 1.0.1, so I don't think changing it to 1.0.3 is correct.
https://www.npmjs.com/package/validate-steuerid

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1 @@
 node_modules
-/dist/
```
Comment (DanaNussair) [inline .gitignore:2]: @UgurGumushan Please commit the dist folder and yarn.lock file.

File: package.json [hunk 3]
```diff
@@ -17,6 +16,8 @@
   "repository": "kontist/validate-steuerid",
   "author": "Kontist GmbH (kontist.com)",
   "main": "dist/index.js",
+  "module": "dist/main.js",
+  "exports": "./dist/main.js",
   "scripts": {
     "transpile": "tsc && babel --root-mode upward src --extensions \".js\" --out-dir dist --copy-files --source-maps --quiet && echo \"--ALL FILES TRANSPILED!\"",
     "test": "ts-mocha -p tsconfig.json src/**/*.spec.ts"
```
Comment (DanaNussair) [inline package.json:23]: @UgurGumushan I get the error ```TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for /home/dananussair/Development/validate-steuerid/src/index.spec.ts``` when I try to run the tests.

**Review by DanaNussair — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (UgurGumushan) [reply package.json:3]: After we merge and you run npm publish it should be updated there. (Maybe I am missing something)

File: .gitignore (hunk 2 above)
Comment (UgurGumushan) [reply .gitignore:2]: [Yes](https://github.com/kontist/validate-steuerid/pull/3/commits/39300d0ae4e4faa371cd4b6a6d4c0615e5920c75)

File: package.json (hunk 3 above)
Comment (UgurGumushan) [reply package.json]: [yarn test should work after this commit](https://github.com/kontist/validate-steuerid/pull/3/commits/39300d0ae4e4faa371cd4b6a6d4c0615e5920c75)

File: package.json (hunk 1 above)
Comment (DanaNussair) [reply package.json:3]: I'm saying that the version should be 1.0.2 instead, cause your code shows that we skipped this version (went from 1.0.1 to 1.0.3 straight)

Comment (UgurGumushan) [reply package.json:3]: Ah yes, I needed to avoid conflicts on my local. Can do 1.0.2

Comment (UgurGumushan) [reply package.json]: [Done](https://github.com/kontist/validate-steuerid/pull/3/commits/5390835d942845dc9384debb8353a9891da30d8b)

File: test/test.js [hunk 4]
```diff
@@ -53,7 +46,7 @@ describe('Generate function', () => {
 
 describe('isOccurencesValid function returs false on obviously false numbers', () => {
```
Comment (DanaNussair) [inline test/test.js:47]: @UgurGumushan a couple of typos. Would you fix them in this PR? 
`returns` and the function name has double R -> `areOccurrencesValid` (should be changed everywhere).

**Review by DanaNussair — APPROVED** [review]

**Review by heriberry — APPROVED** [review]

**Pull request merged**